ksqldb-functional-tests/src/test/resources/rest-query-validation-tests/pull-queries-with-range-scan.json

```diff
@@ -0,0 +1,354 @@
+{
+  "comments": [
+    "Tests covering pull queries with range scan"
+  ],
+  "tests": [
+    {
+      "name": "range scan query must fail with table scan and range scan disable",
+      "statements": [
+        "CREATE TABLE INPUT (ID STRING PRIMARY KEY, V0 STRING, V1 STRING) WITH (kafka_topic='test_topic', value_format='JSON');",
+        "CREATE TABLE MATVIEW AS SELECT ID, V0, V1 FROM INPUT;",
+        "SELECT ID, V0, V1 FROM MATVIEW WHERE ID <= '10';"
+      ],
+      "properties": {
+        "ksql.query.pull.table.scan.enabled":  false,
+        "ksql.query.pull.range.scan.enabled":  false
+      },
+      "expectedError": {
+        "message": "Query requires table scan to enabled"
```

Comment:
could we be smart here and suggest enabling range scans in the case when the comparison would otherwise support range scans? this can be a follow-up ticket

Reply:
not sure how do this in a single validation test. for those few tests where we force range queries to fall back to table scans we can't determine if they could in theory use the range optimization, that's why we have those tests paired with a test that enables range scan.